tools/federation/src/main/java/org/eclipse/rdf4j/federated/evaluation/iterator/GraphToBindingSetConversionIteration.java

```diff
@@ -68,4 +68,8 @@ protected BindingSet convert(Statement st) {
 		return result;
 	}
 
+	@Override
+	protected void handleClose() throws QueryEvaluationException {
+		graph.close();
+	}
```

Comment:
The graph object was not being closed, which is what was causing my original issues.